egs/gale_arabic/s5b/local/prepare_lm.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Copyright 2012  Vassil Panayotov
+#           2017  Ewald Enzinger
+# Apache 2.0
+
+. ./path.sh || exit 1
+
+echo "=== Building a language model ..."
+
+locdata=data/local/lm/
+mkdir -p $locdata
+
+# Language model order
+order=3
+
+. utils/parse_options.sh
+
+# Prepare a LM training corpus from the transcripts
+mkdir -p $locdata
+
+loc=`which ngram-count`;
+if [ -z $loc ]; then
+  if uname -a | grep 64 >/dev/null; then # some kind of 64 bit...
+    sdir=$KALDI_ROOT/tools/srilm/bin/i686-m64 
+  else
+    sdir=$KALDI_ROOT/tools/srilm/bin/i686
+  fi
+  if [ -f $sdir/ngram-count ]; then
+    echo Using SRILM tools from $sdir
+    export PATH=$PATH:$sdir
+  else
+    echo You appear to not have SRILM tools installed, either on your path,
+    echo or installed in $sdir.  See tools/install_srilm.sh for installation
+    echo instructions.
+    exit 1
+  fi
+fi
+
+cat data/train/text | cut -d " " -f 2- >  $locdata/train.txt
+
+ngram-count -text $locdata/train.txt -order $order -interpolate \
+  -kndiscount -lm $locdata/lm.gz
```

Comment:
This doesn't seem the same as the kaldi_lm setup w.r.t. the treatment of vocabulary.  The LM estimation differences should be quite small; it may be to do with the treatment of the vocabulary and of `<UNK>`.    In the wsj setup look at wsj_train_lms.sh.  See these options:
 ```
-limit-vocab -vocab $sdir/wordlist.final.s -unk \
  -map-unk "<UNK>" 
```
I don't know if some subset of these might work as well, but anyway, I assume that would do something similar to what kaldi_lm is doing.  Check, after you build the LMs that the prob of unk is similar vs. kaldi_lm, and that unk appears in a similar number of n-gram contexts.  And check how frequently unk is decoded in the two setups.
I'm assuming this is a phonetic system and that you are still using the QCRI lexicon.